replay_processing/sc2files.py

```diff
@@ -95,10 +94,11 @@ def extract_player_data(result, player, logger, debug=False):
     return data
 
 
-def dump(data):
+def dump(data, outfile):
```

Comment:
This shouldnt be an issue - I added a lock around the writing and we actually will be faster writing the data as it becomes available (another thread will just steal the i/o / cpu while one blocks). I dont think its wrong necessarily to perform dumping as a final step, but not needed ATM AFAICT.